zml/tensor.zig

```diff
@@ -159,9 +159,33 @@ pub const Tensor = struct {
     }
 
     pub fn withSharding(self: Tensor, axes_: anytype) Tensor {
-        var res = self;
-        res._shape = self._shape.withSharding(axes_);
-        return res;
+        return switch (self._id) {
+            .mlir => {
+                var res = self;
+                res._shape = self._shape.withSharding(axes_);
+                return res;
+            },
+            .arg_id => {
```

Comment:
I would have thought this case to be unreachable cause normally `.arg_id` are a temporary creation inside `compileInternal`

Reply:
Yes as you said forward args aka `tensor_args` are mapped with `arg_id` and we even if we already have the sharding info (inherited from the shape received from `zml.compileModel`) we need to hint the compiler this time, `a` and `b` from `Benchmark` are `arg_id`s actually)